src/shipper/log.funcs/extract.json.ts

```diff
@@ -25,9 +25,10 @@ export function onLogExtractJson(lo: LogObject): boolean | void {
 
   const extracted = extractJson(lo.message);
   if (extracted == null) return;
+  const hasMessageField = 'message' in extracted;
```

Comment:
NIT: you could just delete lo.message before the `Object.assign` since then it will be repopulated with the assign.

Reply:
Good point, I like that a lot better.